core/embed/rust/src/ui/model_mercury/bootloader/mod.rs

```diff
@@ -0,0 +1,325 @@
+                fingerprint.into(),
+                button_bld_menu(),
+            );
+
+        if let Some(alert) = alert {
+            frame = frame.with_alert(alert);
+        }
+
+        run(&mut frame)
+    }
+
+    fn screen_wipe_confirm() -> u32 {
+        let icon = Icon::new(FIRE40);
+
+        let msg = Label::centered(
+            "Are you sure you want to factory reset the device?".into(),
+            TEXT_WIPE_NORMAL,
+        );
+        let alert = Label::centered("SEED AND FIRMWARE\nWILL BE ERASED!".into(), TEXT_WIPE_BOLD);
+
+        let right = Button::with_text("RESET".into()).styled(button_wipe_confirm());
+        let left = Button::with_text("CANCEL".into()).styled(button_wipe_cancel());
+
+        let mut frame = Confirm::new(BLD_WIPE_COLOR, left, right, ConfirmTitle::Icon(icon), msg)
+            .with_alert(alert);
+
+        run(&mut frame)
+    }
+
+    fn screen_unlock_bootloader_confirm() -> u32 {
+        unimplemented!();
+    }
+
+    fn screen_unlock_bootloader_success() {
+        unimplemented!();
+    }
+
+    fn screen_menu(firmware_present: secbool) -> u32 {
+        run(&mut Menu::new(firmware_present))
+    }
+
+    fn screen_intro(bld_version: &str, vendor: &str, version: &str, fw_ok: bool) -> u32 {
+        let mut title_str: BootloaderString = String::new();
+        unwrap!(title_str.push_str("BOOTLOADER "));
+        unwrap!(title_str.push_str(bld_version));
+
+        let mut version_str: BootloaderString = String::new();
+        unwrap!(version_str.push_str("Firmware version "));
+        unwrap!(version_str.push_str(version));
+        unwrap!(version_str.push_str("\nby "));
+        unwrap!(version_str.push_str(vendor));
+
+        let mut frame = Intro::new(
+            title_str.as_str().into(),
+            version_str.as_str().into(),
+            fw_ok,
+        );
+
+        run(&mut frame)
+    }
+
+    fn screen_boot_stage_1(fading: bool) {
+        if fading {
+            ModelMercuryFeatures::fadeout();
+        }
+
+        display::rect_fill(SCREEN, BLACK);
+
+        let mut frame = WelcomeScreen::new();
+        show(&mut frame, false);
+
+        if fading {
+            ModelMercuryFeatures::fadein();
+        } else {
+            display::set_backlight(BACKLIGHT_NORMAL);
+        }
+        display::refresh();
+    }
+
+    fn screen_wipe_progress(progress: u16, initialize: bool) {
+        ModelMercuryFeatures::screen_progress(
+            "Resetting Trezor",
+            progress,
+            initialize,
+            BLD_FG,
+            BLD_WIPE_COLOR,
+            Some((Icon::new(FIRE32), BLD_FG)),
+            None,
+        )
+    }
+
+    fn screen_install_progress(progress: u16, initialize: bool, initial_setup: bool) {
+        let bg_color = if initial_setup { WELCOME_COLOR } else { BLD_BG };
+        let fg_color = if initial_setup { GREEN_LIGHT } else { BLD_FG };
+        let icon_color = BLD_FG;
+
+        ModelMercuryFeatures::screen_progress(
```

Comment:
this should be `Self::xyz` in all usages (inside `impl [X for] ModelMercuryFeatures`)

(same holds for the other models of course)